src/components/StopWatch.tsx

```diff
@@ -1,26 +1,33 @@
 import { useEffect, useState } from "react";
+import { Button } from "./Button";
 import { Timer } from "./Timer";
 export const StopWatch = () => {
   const [time, setTime] = useState(0);
-  const [running, setRunning] = useState(false);
+  const [status, setStatus] = useState("");
   useEffect(() => {
     let interval: any;
-    if (running) {
+    if (status === "started") {
       interval = setInterval(() => {
         setTime((prevTime) => prevTime + 10);
       }, 10);
-    } else if (!running) {
+    } else if (status === "paused") {
       clearInterval(interval);
+    } else {
+      setTime(0);
     }
     return () => clearInterval(interval);
-  }, [running]);
+  }, [status]);
   return (
     <div className="stopwatch">
       <Timer time={time} />
       <div className="buttons">
-        <button onClick={() => setRunning(true)}>Start</button>
-        <button onClick={() => setRunning(false)}>Stop</button>
-        <button onClick={() => setTime(0)}>Reset</button>
+        <Button
+          title={status ? "pause" : "start"}
+          onClick={
+            time ? () => setStatus("paused") : () => setStatus("started")
+          }
+        />
+        <Button title="reset" onClick={() => setStatus("reset")} />
```

Comment:
I don't see a reason to add this component at the moment. This is probably a premature, needless refactor.

**Remember [YAGNI (You aren't gonna need it)](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)**

Try to follow these programming principles in this order

YAGNI > SOLID > DRY